### PR TITLE
Fix issue: Array item properties are created with the wrong indent #h…

### PR DESCRIPTION
**Pull request by andxu**
fix:  https://github.com/redhat-developer/vscode-yaml/issues/192

Signed-off-by: andxu <andxu@microsoft.com>

### CODE REVIEW

**Comment by andxu** [comment]
@JPinkney could you please take a look at this PR?

File: test/autoCompletion5.test.ts [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+        });
+    });
+});
+
+function is_EOL(c) {
+    return (c === 0x0A/* LF */) || (c === 0x0D/* CR */);
+}
+
+function completionHelper(document: TextDocument, textDocumentPosition) {
```
Comment (JPinkney) [inline test/autoCompletion5.test.ts]: If I'm not mistaken you shouldn't need this anymore, it now comes standard inside of the [YAMLCompletion]( https://github.com/redhat-developer/yaml-language-server/blob/master/src/languageservice/services/yamlCompletion.ts#L65)

File: test/autoCompletion5.test.ts [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                completion.then(function (result) {
+                    assert.equal('helm:\n \t             \tname: $1', result.items[0].insertText);
+                }).then(done, done);
+            });
+
+        });
+    });
+});
+
+function is_EOL(c) {
```
Comment (JPinkney) [inline test/autoCompletion5.test.ts]: Same with this

File: src/languageservice/services/yamlCompletion.ts [hunk 3]
```diff
@@ -227,10 +227,21 @@ export class YAMLCompletion {
...
                         if (!propertySchema.deprecationMessage && !propertySchema['doNotSuggest']) {
+                            let identCompensation = '';
+                            if (node.parent && node.parent.type === 'array') {
+                                // because there is a slash '-' to prevent the properties generated to have the correct
+                                // indent
+                                const sourceText = document.getText();
+                                const indexOfSlash = sourceText.lastIndexOf('-', node.start - 1);
+                                if (indexOfSlash > 0) {
+                                    // add one space to compensate the '-'
+                                    identCompensation = ' ' +  sourceText.slice(indexOfSlash + 1, node.start);
```
Comment (JPinkney) [inline src/languageservice/services/yamlCompletion.ts]: Seems to be trailing whitespace here

**Pull request merged**